Fixed Issues hopefully

mytopdog opened the pull request.

YoussefElshemi commented on src/Classes/Invite.js:
```diff
@@ -1,13 +1,24 @@
+  delete() {
+    return new Promise((res, rej) => {
+      request.req('DELETE', `/invites/${this.code}`, {}, this.client.token).then(i => {
+        var invite = new Invite(i);
```
You forget to pass the client object to the class

mytopdog replied:
oh rip

mytopdog replied:
will fix

mytopdog replied:
Fixed :+1:

YoussefElshemi commented on src/Classes/Guild.js:
```diff
@@ -170,7 +171,18 @@ class Guild {
+        let ret;
+
+        if (c instanceof TextChannel) {
+          ret = new TextChannel(this.client.channel_methods().fromRaw(c), this.client);
```
these could be one lines

mytopdog replied:
Possibly I'll make a new commit doing that soon

mytopdog replied:
Ok done now :+1:

YoussefElshemi commented on src/Classes/Guild.js:
```diff
@@ -170,7 +171,18 @@ class Guild {
         topic: (type === 'text' && opt && opt.topic) || null
       }, this.client.token).then(c => {
-        setTimeout(res, 100, res(this.client.channel_methods().fromRaw(c)));
+        const TextChannel = require('./TextChannel');
```
So you don't require multiple times, require outside of the class

mytopdog replied:
Can't do that, it causes error:
`Class extends value #<Object> is not a constructor or null`

YoussefElshemi replied:
Ok

mytopdog commented:
```js
Client.on('message', message => {
  if (message.content.startsWith('!create')) {
    message.guild.createChannel(message.content.split(' ')[1], 'text').then(c => {
      c.send('Created #' + message.content.split(' ')[1]);
    }).catch(console.error);
  }
});
```
Code to test for issue regarding channels not caching guilds (I think, that's what I've gathered so far)
